grants/search.js

```diff
@@ -298,7 +282,12 @@ async function buildMatchCriteria(queryParams) {
      */
     if (queryParams.localAuthority) {
         match.$and.push({
-            'beneficiaryLocation.geoCode': queryParams.localAuthority
+            beneficiaryLocation: {
+                "$elemMatch" : {
+                    geoCode: queryParams.localAuthority,
+                    geoCodeType: GEOCODE_TYPES.localAuthority
+                }
```

Comment:
This change ensures that we'll only get results where the matching geocode is actually of the type requested. Right now on production you can request `?localAuthority=E14000563` and see results, even though `E14000563` is a constituency code. This was highlighted due to the dodgy data import of the newest set.

Reply:
💯